cogs/hideout/pits.py

```diff
@@ -315,38 +322,43 @@ async def pit_archive(self, ctx: HideoutGuildContext, *, channel: discord.TextCh
         else:
             await ctx.send(f'✅ **|** Archived **{pit.name}**')
 
-    @counselor_only()
     @pit.command(name='unarchive', with_app_command=False)
     async def pit_unarchive(self, ctx: HideoutGuildContext, *, channel: discord.TextChannel = commands.CurrentChannel):
         """Archives a pit."""
-
         record = await ctx.bot.pool.fetchrow('''SELECT * FROM pits WHERE pit_id = $1''', channel.id)
+
         if record is None:
             raise commands.BadArgument('Could not find pit')
 
         owner = await self.bot.get_or_fetch_member(ctx.guild, record['pit_owner'])
+        pit = ctx.guild.get_channel(record["pit_id"])
+        pits_category = ctx.guild.get_channel(PIT_CATEGORY)
+
         if owner is None:
             raise commands.BadArgument('Could not find pit owner from id')
+        elif pit is None or not isinstance(pit, discord.TextChannel):
+            raise commands.BadArgument('Could not find pit from id')
+        elif pits_category is None or not isinstance(pits_category, discord.CategoryChannel):
+            raise commands.BadArgument('Could not find valid pit category')
 
-        try:
-            pit: discord.TextChannel | None = ctx.guild.get_channel(record["pit_id"])  # type: ignore
-            if pit is None:
-                raise commands.BadArgument('Could not find pit from id')
+        archive_mode = ArchiveMode(record['archive_mode'])
+        is_not_counselor = ctx.guild.get_role(COUNSELORS_ROLE) not in ctx.author.roles
 
-            pits_category: Optional[discord.CategoryChannel] = ctx.guild.get_channel(PIT_CATEGORY)  # type: ignore
-            if pits_category is None:
-                raise commands.BadArgument('Could not find pit category')
+        if archive_mode is ArchiveMode.MANUAL and ctx.author != owner or is_not_counselor:
+            raise ActionNotExecutable('This pit was manually archived, only the pit owner and counsellors can unarchive it.')
+        elif archive_mode is ArchiveMode.INACTIVE and is_not_counselor:
+            raise ActionNotExecutable('This pit was marked as inactive, only the counsellors can unarchive it.')
```

Comment:
This logic should be the other way around, manual (archived by a counsellor) should not permit users from unarchiving it. Users should only be able to unarchive their own pit if it's been archived automatically.

```suggestion
        if archive_mode is ArchiveMode.INACTIVE and ctx.author != owner or is_not_counselor:
            raise ActionNotExecutable('This pit was marked as inactive, only the pit owner and counsellors can unarchive it.')
        elif archive_mode is ArchiveMode.MANUAL and is_not_counselor:
            raise ActionNotExecutable('This pit was manually archived, only the counsellors can unarchive it.')
```

Reply:
Swapped!